src/compiler/checker.ts

```diff
@@ -37830,6 +37830,12 @@ namespace ts {
             return checkVariableLikeDeclaration(node);
         }
 
+        function checkOmittedExpression(node: OmittedExpression) {
+            if (node.parent.parent?.parent?.parent.kind === SyntaxKind.ForOfStatement) {
+                checkRightHandSideOfForOf(node.parent.parent.parent.parent as ForOfStatement);

```

Comment:
`checkVariableLikeDeclaration` already has logic for evaluating the right side of the for-of loop (when encountering a *binding pattern*) when there are no elements. I think it might make more sense to update that code to account for omitted expressions?

Reply:
Hi @Fireboltofdeath, thank you for your review!

> checkVariableLikeDeclaration already has logic for evaluating the right side of the for-of loop (when encountering a binding pattern) when there are no elements. I think it might make more sense to update that code to account for omitted expressions?

This is good point. I did try to update that piece of code to  account for `OmittedExpression`, however here is what I encountered:
- In the logic that you referred to, `checkRightHandSideOfForOf` is called _inside_ `getWidenedTypeForVariableLikeDeclaration` by `getTypeForVariableLikeDeclaration` almost like a side-effect.
- Since our fix is really "check RHS of `for..of` statement _too_ if a node is a  `BindingPattern`, has no `BindingElement`, but does have `OmittedExpression(s)`", it seems a kind of a stretch to fit our fix into logic that check `WidenedType`.

So it seemed more straightforward to "if we encounter `OmittedExpression` in `checkSourceElementWorker`, check RHS of `For..Of` Statement if it is in one".

An alternative approach to work with `checkVariableLikeDeclaration` maybe:
if `node.name.elements > 0` and `node.name.elements` does _not_ contain `BindingElement`, do a check of RHS of `For..Of`.

What do you think? I would love to know you comment on this.

Thanks a lot for your time!

Reply:
I think `needsCheckWidenedType` just needs to be updated so that it’s `true` if all elements are OmittedExpressions; let me give you that since the best expression of it uses several utilities you’re probably not aware of: `!some(node.name.elements, not(isOmittedExpression))`.

You’ll also want to add a couple more test cases:

```ts
for (const [,] of undefined) {
}

for (const [,] of []) {
}
```

This will ensure that we’re also checking that the RHS is iterable, and that its iterated type is also iterable. (Both of these should be errors.)

Thanks!

Reply:
Thank you, I will work on a new commit